My Recipes/scr/Controllers/AddRecipeViewController.swift

```diff
@@ -0,0 +1,316 @@
+            break
+        }
+        
+    }
+}
+
+// MARK: - Selectors
+extension AddRecipeViewController{
+    @IBAction func servesStepperValueChanged(){
+        let people = Int(servesStepper.value)
+        lblNoOfServes.text = "\(people) \(people == 1 ? "People" : "Peoples")"
+    }
+    
+    @objc func segmentValueChanged(_ sender:UISegmentedControl){
+        complexityIndex = sender.selectedSegmentIndex
+    }
+    
+    @objc func saveTapped(){
+        let params: [String:Any] = [WSParams.name: recipeName ?? "",
+                                    WSParams.preparationTime: duration ?? "",
+                                    WSParams.serves: Int(serves),
+                                    WSParams.complexity: complexityArray[complexityIndex]]
+        indicator.startAnimation()
+        recipeProvider.request(.add(params: params)) { result in
+            self.indicator.stopAnimation()
+            switch result{
+            case let .success(response):
+                do{
+                    struct Response: Codable {
+                        let msg: String?
+                        let id: Int?
+                    }
+                    let info = try JSONDecoder().decode(Response.self, from: response.data)
+                    if let message = info.msg{
+                        Toast(text: message).show()
+                    }
+                    self.navigationItem.rightBarButtonItem?.isEnabled = false
+                    self.recipeId = info.id
+                    print(info)
+                    self.tableView.beginUpdates()
+                    self.tableView.endUpdates()
+                }catch let error{
+                    print(error)
+                }
+                break
+            case let .failure(error):
+                print("failure",error)
+                break
+            }
+        }
+    }
+    
+    private func setRecipeImage(){
+        guard let photo = recipePhoto, let recipeId = recipeId else {
+            return
+        }
+        indicator.startAnimation()
+        recipeProvider.request(.addRecipePhoto(recipeId: recipeId, photo: photo)) { result in
+            self.indicator.stopAnimation()
+            switch result{
+            case let .success(response):
+                do{
+                    struct Response: Codable {
+                        let msg,photo: String?
+                    }
+                    let info = try JSONDecoder().decode(Response.self, from: response.data)
+                    if let message = info.msg{
+                        Toast(text: message).show()
+                    }
+                }catch let error{
+                    print(error)
+                }
+                break
+            case let .failure(error):
+                print("failure",error)
+                break
+            }
+        }
+    }
+}
+
+extension AddRecipeViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate{
```

Comment:
Extensions should contain only one delegate methods at a time for better readability and code understanding.

Reply:
To set UIImagePickerController delegate, must need to extends UINavigationControllerDelegate but don't need to override any method of UINavigationControllerDelegate.